exe/audio-visualizer-back-end

```diff
@@ -21,17 +21,21 @@ if data_file.nil? && interval < 60
   exit 1
 end
 
-metrics = Metrics::Loader.new(data_file: data_file).load(METRICS_FILE)
-
-until metrics.all?(&:done?)
-  # Load the data from Datadog or a file.
-  metrics.reject(&:done?).each do |metric|
-    if data_file
-      # The contents of the file are in the past, so we have to fake it.
-      metric.load(Time.at(0), Time.at(2**32))
-    else
-      metric.load(Time.now - interval, Time.now)
-    end
+
+definitions = JSON.parse(File.read(METRICS_FILE))
+
+if data_file
+  stream = Metrics::FileStream.new(definitions, data_file)
+else
+  stream = Metrics::DatadogStream.new(definitions)
+end
+
+
+until stream.done?
+  if data_file
+    metrics = stream.read_data(Time.at(0), Time.at(2**32))
```

Comment:
Perhaps we should leave the comment here to explain the otherwise opaque `2**32` thing.